Add .load(id) method to JpaRepository

mitchjust-finocomp opened the pull request:
Resolves https://github.com/micronaut-projects/micronaut-data/issues/893

Any advice for safer ways to extract the parameter from the `MethodInvocationContext` would be much appreciated!

graemerocher commented:
looks ok, might want a `null` check since loading with a null ID is likely to produce an error?

mitchjust-finocomp commented:
@graemerocher the null check was already being performed, I've added a test to confirm that we get a `ConstraintViolationException`. Still, I've added some extra safety to extracting the parameter.

graemerocher commented:
Thanks for the contribution!
